server/src/main/java/io/whitefox/persistence/StorageManager.java

```diff
@@ -3,13 +3,16 @@
 import io.whitefox.api.deltasharing.model.Schema;
 import io.whitefox.api.deltasharing.model.Share;
 import io.whitefox.api.deltasharing.model.Table;
+import io.whitefox.persistence.memory.PTable;
 import java.util.List;
 import java.util.Optional;
 import java.util.concurrent.CompletionStage;
 
 public interface StorageManager {
   CompletionStage<Optional<Share>> getShare(String share);
 
+  CompletionStage<Optional<PTable>> getPTable(String share, String schema, String table);
```

Comment:
As discussed on the other PR with @duhizjame probably we should refactor the StorageManager to return P classes instead of API ones